go/vt/vtgate/planbuilder/operators/subquery.go

```diff
@@ -208,7 +208,7 @@ func (sq *SubQuery) GetMergePredicates() []sqlparser.Expr {
 }
 
 func (sq *SubQuery) settle(ctx *plancontext.PlanningContext, outer Operator) Operator {
-	if !sq.TopLevel {
+	if !sq.TopLevel && sq.correlated {
```

Comment:
Can you explain why this is needed?

Reply:
We want to allow uncorrelated queries when subquery isn't the top level construct, like its underneath an Aggregator, and we can allow uncorrelated queries because they will be pulled out and run separately.

Reply:
I'll add a comment to the code too for ☝️

Reply:
Got it, thank you

Reply:
Added the comment